### PR TITLE
[html] Fix indentation when entering a newline while in the middle of a tag

### PR DESCRIPTION
**Pull request by JJK96**
# Example:
    
```
        <tag>|</tag>
```
Then press enter
    
## Old behaviour:
```
        <tag>
            </tag>
```
## New behaviour:
```
        <tag>
        </tag>
```

I also added a test for it

### CODE REVIEW

**Comment by JJK96** [comment]
Sry forgot to rebase, I'm on it

File: rc/filetype/html.kak [hunk 1]
```diff
@@ -18,7 +18,7 @@ hook global BufCreate .*\.xml %{
 hook global WinSetOption filetype=(html|xml) %{
     require-module html
 
-    hook window ModeChange pop:insert:.* -group "%val{hook_param_capture_1}-trim-indent"  html-trim-indent
+    hook window ModeChange insert:.* -group "%val{hook_param_capture_1}-trim-indent"  html-trim-indent
```
Comment (Delapouite) [inline rc/filetype/html.kak]: pop is needed in the latest builds of kakoune

File: rc/filetype/html.kak [hunk 2]
```diff
@@ -77,11 +77,11 @@ define-command -hidden html-indent-on-greater-than %[
 define-command -hidden html-indent-on-new-line %{
     evaluate-commands -draft -itersel %{
         # preserve previous line indent
-        try %{ execute-keys -draft <semicolon> K <a-&> }
+        try %{ execute-keys -draft \; K <a-&> }
```
Comment (Delapouite) [inline rc/filetype/html.kak]: same, `<semicolon>` should stay

**Comment by JJK96** [comment]
I overwrote the file without looking at the changes, only meant to change 2 lines.

**Pull request merged**